addons/music/fnc_isMusicPlaying.sqf

```diff
@@ -20,19 +20,38 @@ Author:
     Dedmen, Commy2, Fishy
 ---------------------------------------------------------------------------- */
 
-#define MINIMUM_WAIT 0.1
+#define WAIT_SWITCH_FPS 10
+#define WAIT_HIGH_FPS 0.6
+#define WAIT_LOW_FPS 3
 
 if (isNil QGVAR(track)) exitWith {false};
 
-GVAR(track) params ["_class", "_startTime", "_playPos", "_duration"];
-
-private _return = true;
-
-// Check if theres a track playing, since `playMusic ""` doesn't trigger an event
-private _trackTime = getMusicPlayedTime;
-if (time > (_startTime + MINIMUM_WAIT) && {_trackTime == 0}) then {
+GVAR(track) params ["_class", "_startTime", "_startPos", "_duration"];
+
+if (getMusicPlayedTime != 0) exitWith {true};
+
+/* Check if there is no track playing, since `playMusic ""` doesn't trigger an event
+ * In order to do so, check if `getMusicPlayedTime` is zero. However `getMusicPlayedTime` seems to be an estimation
+ * and can result in the same value over multiple milliseconds, leading to false positives if the track just started
+ * In order to prevent this, this function always waits a minimum amount of time as to allow `getMusicPlayedTime` to update
+ * The ideal wait time is FPS depended, so in order to keep it simple and still responsive there are two wait times
+ * A high FPS wait time and a low FPS wait time, for which the values are based of the following data:
+
+ * | FPS | Max Time to Update | Average Time to Update |
+ * | --- | ------------------ | ---------------------- |
+ * | 2   | 2.333010           | 1.155892               |
+ * | 10  | 0.385986           | 0.300724               |
+ * | 20  | 0.240967           | 0.191904               |
+ * | 36  | 0.170898           | 0.157701               |
+ * | 59  | 0.150009           | 0.126583               |
+ * | 120 | 0.120972           | 0.106157               |
+ * | --- | ------------------ | ---------------------- |
+ * This data was collected from 2 machines on a locally hosted server with only one player */
+private _miniumWaitTime = [WAIT_HIGH_FPS, WAIT_LOW_FPS] select (diag_fpsMin < WAIT_SWITCH_FPS);
```

Comment:
This is just weird. I hate it.

Reply:
Could you elaborate what it is that you hate about this? Its implementation or the the solution it self?

Also I just noticed an over sight on my part, this was meant to be `diag_fps` instead of `diag_fpsMin`

Reply:
It just does not make any sense to have code run differently on different framerate.

Reply:
Got it, then I'll change it to accommodate for worst case scenario.

Reply:
Got it, I'll change it one minimum wait time that accommodates the worst case scenario.